man/man8/zpoolconcepts.8

```diff
@@ -93,6 +93,56 @@ compromised.
 The minimum number of devices in a raidz group is one more than the number of
 parity disks.
 The recommended number is between 3 and 9 to help increase performance.
+.It Sy draid , draid1 , draid2 , draid3
+A distributed version of raidz with integrated hot spares that allows for
+faster resilvering.
+A draid vdev is constructed from multiple internal raidz groups, each with D
+data devices and P parity devices, which are distributed over the children.
+This allows all of the children to participate when resilvering.
+As a general rule the more devices in a draid the less time required to
+resilver to a distributed spare.
+Furthermore, a narrower stripe width (D+P) will improve resilver performance
+at the expense of total usable capacity.
+.Pp
+Like raidz a draid can have single-, double-, or triple-parity.  The
+.Sy draid1 ,
+.Sy draid2 ,
+and
+.Sy draid3
+types can be used to specified the required parity level.
+The
+.Sy draid
+vdev type is an alias for
+.Sy draid1 .
+.Pp
+A draid with N disks of size X, D data disks, P parity level, and H distributed
+hot spares can hold approximately (N-H)*(D/(D+P))*X bytes and can withstand P
+device(s) failing before data integrity is compromised.
+The minimum number of devices in a draid is P+2.
+.It Sy draid[<parity>][:<data>d][:<children>c][:<spares>s][:<iterations>i]
+When a non-default draid configuration is required it can be specified by
+appending one or more of the following optional arguments to the
+.Sy draid
+keyword.
+.Pp
+.Em children
+- The expected number of children.
+If the provided number of children is different an error is returned.
+.Pp
+.Em data
+- The number of data devices per redundancy group.
+.Pp
+.Em iterations
+- Iterations to perform when developing the permutation map.
+.Pp
+.Em parity
+- The parity level as described above (1-3).
+.Pp
+.Em spares
+- The number of distributed hot spares.
+Every pool must contain at least one but may contain more.
```

Comment:
```suggestion
Every draid group must contain at least one but may contain more.
```

Would it be possible to allow zero distributed spares?  I think that could make draid applicable in more use cases as a replacement for raidz.

Reply:
This was correct.  The distributed spare is for the pool not the draid group.  Yes, I think we could probably allow zero distributed spares.  Though the core advantage of draid is the distributed resilver time which a zero spare configuration of course wouldn't be able to take advantage of.  However, it would still be able to perform sequential resilvers, unlike raidz, and there may be other performance advantages to supporting it.  I'll look in what's required.

Reply:
Oh interesting, so you could have a pool with a mirror (e.g. `special mirror`) plus a `draid`, and if a disk in the mirror dies, we can replace it with the "distributed spare" from the draid?  That's neat.

Reply:
Updated PR to no longer requires a distributed spare.  Not quite, the distributed spare is only usable by the top-level dRAID it's part of.